tests/test_buku.py

```diff
@@ -547,20 +547,23 @@ def test_is_nongeneric_url(url, exp_res):
     assert res == exp_res
 
 
-@pytest.mark.parametrize(
-    "newtag, exp_res",
-    [
-        (None, ("http://example.com", "text1", ",", None, 0, True, False)),
-        ("tag1", ("http://example.com", "text1", ",tag1,", None, 0, True, False)),
-    ],
-)
-def test_import_md(tmpdir, newtag, exp_res):
+@pytest.mark.parametrize('tags, newtag, exp_tags', [
+    ('', None, ','), ('', 'new tag', ',new tag,'),
+    ('foo, bar, baz', None, ',bar,baz,foo,'),
+    ('foo, bar, baz', 'new tag', ',bar,baz,foo,new tag,'),
+])
+@pytest.mark.parametrize('title', ['Bookmark title', '', None])
+@pytest.mark.parametrize('url', ['http://example.com', 'javascript:void(0)', ''])
+def test_import_md(tmpdir, url, title, tags, newtag, exp_tags):
     from buku import import_md
 
     p = tmpdir.mkdir("importmd").join("test.md")
-    p.write("[text1](http://example.com)")
+    print(line := (f'<{url}>' if title is None else f'[{title}]({url})') +
+                  ('' if not tags else f' <!-- TAGS: {tags} -->'))
```

Comment:
Printing out the line to be parsed makes it easier to figure out what went wrong when a test fails.